Adapter.C

```diff
@@ -462,7 +459,8 @@ void preciceAdapter::Adapter::execute()
     ACCUMULATE_TIMER(timeInWriteResults_);
 
     // Read the received coupling data from the buffer
-    readCouplingData();
+    // Fits to an implicit Euler
+    readCouplingData(runTime_.deltaT().value());
```

Comment:
Rubber-ducking, to cross-check that this is what you intended:

1. OpenFOAM computes a new time step size (which may be modified by the adapter) (let's call it **dt1**)
2. OpenFOAM computes a new solution with this time step size
3. The adapter writes data, calls `advance()`, and takes care of checkpointing
4. In case there is a fixed time step size, the adapter resets it to the value from the configuration (let's call it **dt2**)
5. The adapter reads data at a step **dt1** (if dynamic dt) or **dt2** (if fixed dt).

Sounds correct for the dynamic time step size, but we may need to check what happens at a fixed time step size. We may be introducing (or already had) a bug here.

Reply:
@davidscn any comment here?